src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Performance/UseSpanClearInsteadOfFill.cs

```diff
@@ -0,0 +1,150 @@
+    /// </summary>
+    [DiagnosticAnalyzer(LanguageNames.CSharp, LanguageNames.VisualBasic)]
+    public sealed class UseSpanClearInsteadOfFillAnalyzer : DiagnosticAnalyzer
+    {
+        internal const string DiagnosticId = "CA1851";
+        internal const string FillMethod = "Fill";
+        internal const string ClearMethod = "Clear";
+
+        private static readonly LocalizableString s_localizableTitle = new LocalizableResourceString(nameof(MicrosoftNetCoreAnalyzersResources.UseSpanClearInsteadOfFillTitle), MicrosoftNetCoreAnalyzersResources.ResourceManager, typeof(MicrosoftNetCoreAnalyzersResources));
+        private static readonly LocalizableString s_localizableMessage = new LocalizableResourceString(nameof(MicrosoftNetCoreAnalyzersResources.UseSpanClearInsteadOfFillMessage), MicrosoftNetCoreAnalyzersResources.ResourceManager, typeof(MicrosoftNetCoreAnalyzersResources));
+        private static readonly LocalizableString s_localizableDescription = new LocalizableResourceString(nameof(MicrosoftNetCoreAnalyzersResources.UseSpanClearInsteadOfFillDescription), MicrosoftNetCoreAnalyzersResources.ResourceManager, typeof(MicrosoftNetCoreAnalyzersResources));
+
+        internal static readonly DiagnosticDescriptor s_Rule = DiagnosticDescriptorHelper.Create(
+            DiagnosticId,
+            s_localizableTitle,
+            s_localizableMessage,
+            DiagnosticCategory.Performance,
+            RuleLevel.IdeSuggestion,
+            description: s_localizableDescription,
+            isPortedFxCopRule: false,
+            isDataflowRule: false);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(s_Rule);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecution();
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+
+            context.RegisterCompilationStartAction(OnCompilationStart);
+        }
+
+        private void OnCompilationStart(CompilationStartAnalysisContext context)
+        {
+            var compilation = context.Compilation;
+            if (!compilation.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemSpan1, out var spanType))
+            {
+                return;
+            }
+
+            var spanFillMethod = spanType.GetMembers(FillMethod)
+                .OfType<IMethodSymbol>()
+                .FirstOrDefault();
+
+            if (spanFillMethod?.Parameters.Length != 1)
+            {
+                return;
+            }
+
+            context.RegisterOperationAction(
+                context =>
+                {
+                    var invocation = (IInvocationOperation)context.Operation;
+
+                    if (SymbolEqualityComparer.Default.Equals(invocation.TargetMethod.OriginalDefinition, spanFillMethod)
+                        && IsDefaultValue(invocation.Arguments[0]))
```

Comment:
Please also test error code where no arguments are provided to `Fill` method call, such as `span.Fill();`. I think this case will not generate an `IInvocationOperation`, but good to have a test to verify it. A more defensive implementation would be to just add an explicit check `invocation.Arguments.Length == 1` before accessing `invocation.Arguments[0]`